### ISSUE TITLE
Fast parser does not produce warnings about tuple argument names

### ISSUE THREAD

**Issue by ddfisher** [python/mypy#2732]
Python 2 tuple arguments may have names which conflict with other arguments.  These are not caught with the fast parser.
For example, this function produces no errors:
```
def f(x, (x, y)):
  # type: (...) -> None
  pass
```

**Comment by afrieder** [comment]
Should a fix to this catch the more general:

```python
def f(x, x):
  pass
```
?
This is currently accepted by both parsers without error.

**Comment by ddfisher** [comment]
That's a red herring -- both parsers are able to catch that error case, but
only do so when the function is annotated.

**Comment by afrieder** [comment]
Hm. That makes sense I guess. It feels odd to ignore that for unannotated functions when 
1. It's so easy to do the check, and
2. The program will immediately crash when run (because syntax error)

Like, I guess it's not mypy's job to do anything but typecheck, but I would expect any linter to do basic syntax checking.

**Comment by ddfisher** [comment]
I didn't look at this carefully enough.  It's not really caught by the fast parser at all -- it just causes an error at a later stage.  However, it *is* a true syntax error in Python.  #2736 is a definite improvement.  Thanks!
